### PR TITLE
bpo-35186: Remove "built with" comment in setup.py upload

### PR DESCRIPTION
**Pull request by pganssle**
platform.dist() is deprecated and slated for removal in Python 3.8. The upload command itself should also not be used to upload to PyPI, but while it continues to exist it should not use deprecated functions.

This is the first option I proposed in [bpo-35186](https://bugs.python.org/issue35186). A slightly more conservative option (option 2) is easy enough to implement:

```diff
diff --git a/Lib/distutils/command/upload.py b/Lib/distutils/command/upload.py
index 32dda359ba..56ba12b1c5 100644
--- a/Lib/distutils/command/upload.py
+++ b/Lib/distutils/command/upload.py
@@ -121,12 +121,9 @@ class upload(PyPIRCCommand):
             'requires': meta.get_requires(),
             'obsoletes': meta.get_obsoletes(),
             }
+
         comment = ''
-        if command == 'bdist_rpm':
-            dist, version, id = platform.dist()
-            if dist:
-                comment = 'built for %s %s' % (dist, version)
-        elif command == 'bdist_dumb':
+        if command in ('bdist_rpm', 'bdist_dumb'):
             comment = 'built for %s' % platform.platform(terse=1)
         data['comment'] = comment
```

Distutils maintainers let me know how you want to go. I'll hold off on the news entry until after we decide which way to go.

https://bugs.python.org/issue35186

### CODE REVIEW

**Comment by pganssle** [comment]
@di @dstufft Any chance you know if the "comment" field is used for anything in warehouse?

**Comment by pganssle** [comment]
I think as long as it's only used as a free-form comment it should be fine to remove it from `distutils`. Anyone who runs into this problem can and should use `twine -c "built for $(some_platform)`" or whatever.

At worst this is going to break people who are parsing the comment field from PyPI for whatever reason, and I suspect that's both exceedingly rare and not really something anyone is eager to support.

**Comment by dstufft** [comment]
Yea, what @di said. It has no semantic meaning, just a freeform whatever field.

Honestly we might even want to consider removing it, I'm not sure it really makes much sense in the modern world where almost all downloads from PyPI are automated tooling and not humans making project by project decisions on what file to download.

**Comment by pganssle** [comment]
@berkerpeksag @encukou @malemburg Are y'all the ones who are moving the deprecation of `platform.dist()` forward? I'm not sure who should review / merge this PR.

File: Lib/distutils/command/upload.py
Comment (merwok) [inline Lib/distutils/command/upload.py:125]: Does it make a difference to send empty string or omit comment altogether?

**Review by merwok — APPROVED** [review]

File: Lib/distutils/command/upload.py
Comment (pganssle) [reply Lib/distutils/command/upload.py:125]: The old version sent an empty string by default, so I kept that behavior, but I just checked what `twine` does, and it does seem to send `''` even when `comment=None` (the default), so I think that `distutils` should do the same.

**Comment by encukou** [comment]
`platform.dist()` was removed because it was misleading – it misidentified distros, and there was no sustainable way to keep it up to date.
Efforts to *use distro names* instead of proper feature detection vs. *keep the identification code up to date* form a vicious cycle I explained in a [`distro` bug](https://github.com/nir0s/distro/issues/221#issue-323823924).

I can't see how this comment is useful (esp. if it's added automatically), so I'd be for removing it.

**Comment by pganssle** [comment]
@encukou @berkerpeksag @merwok Anyone want to push the merge button?

**Comment by merwok** [comment]
@encukou Not sure if your last message means «+1 on this PR» or «please change something in the PR»

**Comment by encukou** [comment]
A +1. Thanks for the reminder!

**Pull request merged**
